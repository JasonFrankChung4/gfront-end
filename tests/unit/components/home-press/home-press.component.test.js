import React from 'react'
import { shallow } from 'enzyme'

import HomePress from 'components/home-press/home-press.component'

const props = {
  backgroundImage: 'test-bg-img',
  quote: 'test-quote',
  attributionUrl: '/test-attribution-url',
  attributionImage: 'test-attribution-img',
  links: [
    { image: 'link-image-1', url: '/link-url-1', alt: 'link-alt-1' },
    { image: 'link-image-2', url: '/link-url-2', alt: 'link-alt-2' },
  ],
}

let wrapper
describe('components - HomePress', () => {
  beforeEach(() => {
    wrapper = shallow(<HomePress {...props} />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render nothing for empty `links`', () => {
    wrapper.setProps({ links: undefined })
    expect(wrapper.find('.press-links li').length).toBe(0)
  })
})