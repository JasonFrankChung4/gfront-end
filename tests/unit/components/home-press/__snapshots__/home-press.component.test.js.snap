// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - HomePress should render correctly 1`] = `
<section
  className="press"
  id="press-section"
>
  <div
    className="banner medium"
    id="press-banner"
    style={
      Object {
        "backgroundImage": "url('test-bg-img')",
      }
    }
  >
    <div
      className="banner-content"
    >
      <div
        className="indented"
      >
        <div
          className="banner-title"
        >
          Press
        </div>
        <div
          className="press-container"
        >
          <div
            className="quote light"
          >
            <blockquote>
              test-quote
            </blockquote>
            <div
              className="attribution"
            >
              <a
                href="/test-attribution-url"
              >
                <img
                  alt="Wired"
                  className="press-logo"
                  src="test-attribution-img"
                />
              </a>
            </div>
          </div>
          <div>
            <hr />
            <ul
              className="press-links"
            >
              <li
                key="link-alt-1"
              >
                <a
                  href="/link-url-1"
                >
                  <img
                    alt="link-alt-1"
                    className="press-logo"
                    src="link-image-1"
                  />
                </a>
              </li>
              <li
                key="link-alt-2"
              >
                <a
                  href="/link-url-2"
                >
                  <img
                    alt="link-alt-2"
                    className="press-logo"
                    src="link-image-2"
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`
